CAN_Logger_with_Autobaud_and_Requests_no_CAN2_AES_CBC_SHA256/CAN_Logger_with_Autobaud_and_Requests_no_CAN2_AES_CBC_SHA256.ino

```diff
@@ -989,46 +1006,140 @@ void setup(void) {
   //EEPROM.put(EEPROM_BRAND_NAME_ADDR,brand_name);
 
   Serial.println("Done.");
-
-
-  Serial.print("Writing to baudRate.txt... ");
-  baudFile.open("baudRate.txt", O_RDWR | O_CREAT | O_AT_END);
-  sprintf(timeString,"%04d-%02d-%02dT%02d:%02d:%02d,%d,%d",year(),month(),day(),hour(),minute(),second(),Can0.baud_rate,Can1.baud_rate);
-  baudFile.print(timeString);
-  baudFile.print(" TU");
-  baudFile.print(logger_name);
-  baudFile.print(current_file);
-  baudFile.print(" IV: ");
-  for (int n = 0; n < sizeof(init_vector); n++){
-    baudFile.print(",0x");
-    if (init_vector[n] < 16){
-      baudFile.print("0");
-    }
-    
-    baudFile.print(init_vector[n],HEX);
-  }
-  baudFile.print(" KEY:");
-    for (int n =0; n < sizeof(aeskey);n++){
-    baudFile.print(",0x");
-    if (aeskey[n] <16){
-      baudFile.print("0");
-    }
-     baudFile.print(aeskey[n],HEX);
-  }
-  baudFile.println("");
-  baudFile.close();
-  Serial.println("Done.");
-
   
   sprintf(file_name_prefix,"%s%s",brand_name,logger_name);
   Serial.print("The filename prefix is ");
   Serial.println(file_name_prefix);
+
+  
   
   recording = true;
   pinMode(POWER_PIN,INPUT_PULLUP);
   attachInterrupt(digitalPinToInterrupt(POWER_PIN), close_binFile, RISING);
 }
 
+
+void write_final_meta_data(){
+  sha256Instance->update(cipher_text,BUFFER_SIZE);
+  sha256Instance->final(hash);
+  delete sha256Instance;
+  Serial.print(",BIN-SHA:");
+  memcpy(&data_file_contents[data_file_index],",BIN-SHA:",9);
+  data_file_index+=9;
+  for (int n = 0; n < sizeof(hash); n++){
+    char hex_digit[3];
+    sprintf(hex_digit,"%02X",hash[n]);
+    Serial.print(hex_digit);
+    memcpy(&data_file_contents[data_file_index],&hex_digit,2);
+    data_file_index+=2;
+  }
+    
+  sha256Instance=new Sha256();
+  sha256Instance->update(data_file_contents, strlen((const char*)data_file_contents));
+  sha256Instance->final(hash);
+  delete sha256Instance;
+  
+  Serial.print(",TXT-SHA:");
```

Comment:
maybe newline instead of comma

Reply:
The original intent was to have multiple lines in one file. Now we are looking at one file per log, so the newline is acceptable. The PC and Server side utilities will need to be updated too.